sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -81,11 +81,12 @@ case class CreateTableAsSelectExec(
     }
 
     Utils.tryWithSafeFinallyAndFailureCallbacks({
+      val schema = query.schema.asNullable
```

Comment:
We don't need to update the schema nullability of  the corresponding logical plans, `CreateTableAsSelect` and `ReplaceTable`, in the analyzer phase? Any reason to directly update the nullability of physical plans?

Reply:
I think it's too much work if we need to transform the logical plan and add an extra Project to change the nullability.

Reply:
I think it would be incorrect to change the logical plan. The behavior of CTAS should be that tables are created with nullable types. The query used by CTAS should not be changed.

Reply:
Ah, I see. Thanks. Looks ok to me.